File: Python/apps/RemoteInteractionDemo.py
import os
import sys

# Get the current script directory
current_dir = os.path.dirname(os.path.abspath(__file__))

# Insert libs path
sys.path.insert(0, os.path.join(current_dir, '..', 'libs'))

# Insert the apps path
sys.path.insert(0, current_dir)

from SIRemote import RemoteAPI

api = RemoteAPI()
server_secret: str = 'k28b1'
url_base: str = 'https://localhost:3003'


def check_for_failure():
    """ Error handling for demo """
    if not api.get_response_status().lower() == "success":
        raise IOError(f"Remote interaction failed: {api.get_response_msg()}")


def print_formated_debug() -> None:
    """Prints the Response packet from the global `api` instance."""

    print("DEBUG | PRINTING FORMATED RESPONSE INFO")
    print(f'RESPONSE MSG: {api.get_response_msg()}')
    print(f'RESPONSE STATUS: {api.get_response_status()}')


def main() -> None:
    """ Demonstration of how to use the RemoteAPI for Python.
    Here we will set our scanner URL, the scanner/server_secret and then call a few command functions.

    :return: None
    """

    try:
        # set the target URL
        api.set_ip(url_base)

        # register the device
        api.send_register(server_secret)
        check_for_failure()

        # request an image
        image = api.send_capture_image_frame()
        image.show()

        # save that image to disk
        api.save_image_to_disk(image)

        # toggle the QR and Stillness requirement
        api.send_stillness_required(False)
        check_for_failure()

        api.send_marker_required(True)
        check_for_failure()

        # set the camera settings
        api.send_set_camera_exposure(10000000)
        check_for_failure()

        api.send_set_camera_sensitivity(800)
        check_for_failure()

        # get the scanner state and
        api.send_get_state()
        check_for_failure()
    except Exception as e:
        print(f"Exception caught during demo: {e}")


if __name__ == "__main__":
    main()

File: Java/apps/src/main/java/ninox360/demo/RemoteInteractionDemo.java
package ninox360.demo;

import ninox360.util.RemoteAPI;

import java.awt.image.BufferedImage;
import java.io.IOException;

/**
 * Java Demonstration Code for the Remote Control API.
 * Sets an IP, completes registration, and sends several Commands to the scanner printing the response from the requests.
 */
public class RemoteInteractionDemo {
    static RemoteAPI remote = new RemoteAPI();
    static String serverSecret = "9n0SQ";
    static String outboundIP = "https://localhost:3003";

    /**
     * Optional Debug output to read the Response out.
     */
    public static void printResponse() {
        System.out.println("Server Response:" + "\n  RESPONSE: " + remote.getResponseStatus() + "\n  MESSAGE: " + remote.getResponseMsg() + "\n");
    }

    /**
     * Error handling block the demonstration code.
     *
     * @throws IOException if the response status is not "success" following a request command.
     */
    public static void checkForFailure() throws IOException {
        //printResponse();
        if (!remote.getResponseStatus().equalsIgnoreCase("success")) {
            throw new IOException("Remote interaction failed: " + remote.getResponseMsg());
        }
    }

    public static void main(String[] args) {
        try {
            // set your server IP, a default of 'https://localhost:3003' is prepopulated.
            remote.setIP(outboundIP); // replace this with the IP displayed on your device.

            // Registration is required prior to any server interaction
            remote.sendRegister(serverSecret);
            checkForFailure();

            BufferedImage image = remote.sendCaptureImageFrame();

            // API utility function for saving a Buffered Image to Disk
            if (!remote.saveImageToDisk(image, "saved_frame.jpg")) {
                throw new IOException("Failed to save Image to disk");
            }

            remote.sendGetState();
            checkForFailure();

            remote.sendStillnessRequired(false);
            remote.sendMarkerRequired(true);

            remote.sendSetCameraExposure(10_000_000);
            checkForFailure();

            remote.sendSetCameraSensitivity(400);
            checkForFailure();

        } catch (IOException e) {
            System.err.println("Error during remote interaction: " + e.getMessage());
        }
    }
}